csharp/sources/Valkey.Glide.InterOp/NativeClient.cs

```diff
@@ -0,0 +1,509 @@
+            fixed (NodeAddress* addressesPtr = addresses)
+            {
+                Native.ConnectionRequest nativeRequest = new()
+                {
+                    client_name = MarshalCollectingString(request.ClientName),
+                    cluster_mode_enabled = request.ClusterMode ? 1 : 0,
+                    connection_retry_strategy = new Native.ConnectionRetryStrategy
+                    {
+                        ignore = !request.ConnectionRetryStrategy.HasValue ? 1 : default,
+                        number_of_retries = request.ConnectionRetryStrategy?.NumberOfRetries ?? default,
+                        exponent_base = request.ConnectionRetryStrategy?.ExponentialBase ?? default,
+                        factor = request.ConnectionRetryStrategy?.Factor ?? default,
+                    },
+                    connection_timeout =
+                        new OptionalU32
+                        {
+                            ignore = !request.ConnectionTimeout.HasValue ? 1 : 0,
+                            value = (uint?)request.ConnectionTimeout?.TotalMilliseconds ?? default
+                        },
+                    otel_span_flush_interval_ms =
+                        new OptionalU64
+                        {
+                            ignore = !request.OpenTelemetrySpanFlushInterval.HasValue ? 1 : 0,
+                            value = (ulong?)request.OpenTelemetrySpanFlushInterval?.TotalMilliseconds ?? default
+                        },
+                    request_timeout =
+                        new OptionalU32
+                        {
+                            ignore = !request.RequestTimeout.HasValue ? 1 : 0,
+                            value = (uint?)request.RequestTimeout?.TotalMilliseconds ?? default
+                        },
+                    inflight_requests_limit = new OptionalU32
+                    {
+                        ignore = !request.InflightRequestsLimit.HasValue ? 1 : 0,
+                        value = request.InflightRequestsLimit ?? default
+                    },
+                    periodic_checks = new Native.PeriodicCheck
+                    {
+                        kind = request.PeriodicChecks?.Kind switch
+                        {
+                            EPeriodicCheckKind.Enabled => Native.EPeriodicCheckKind.Enabled,
+                            EPeriodicCheckKind.Disabled => Native.EPeriodicCheckKind.Disabled,
+                            EPeriodicCheckKind.ManualInterval => Native.EPeriodicCheckKind.ManualInterval,
+                            null => Native.EPeriodicCheckKind.None,
+                            _ => throw new ArgumentOutOfRangeException(),
+                        }
+                    },
+                    protocol = request.Protocol switch
+                    {
+                        EProtocolVersion.Resp2 => Native.EProtocolVersion.RESP2,
+                        EProtocolVersion.Resp3 => Native.EProtocolVersion.RESP3,
```

Comment:
We could avoid duplicating types/classes for such simple cases, what do you think?

Reply:
I’m a bit undecided on this one (as you can probably tell from parts of the code).
Initially, I duplicated the enums to avoid relying on native interop, but in hindsight, it introduced a lot of unnecessary complexity. I’m open to reducing the duplication, really just call a direction here.